### PR TITLE
Update gocbcorex and use bucket tracking agent manager

### PR DESCRIPTION
**Pull request by chvck**
This change addresses ING-569 ING-535

### CODE REVIEW

**Comment by chvck** [comment]
Surprised how easily this dropped in...

**Review by brett19 — APPROVED** [review]
The design structure of gocbcorex works!

File: gateway/dataimpl/server_v1/bucketadminserver.go [hunk 1]
```diff
@@ -193,18 +193,18 @@ func (s *BucketAdminServer) CreateBucket(
...
+				ReplicaNumber:      numReplicas,
+				EvictionPolicy:     evictionPolicy,
+				MaxTTL:             maxExpiry,
+				CompressionMode:    compressionMode,
+				DurabilityMinLevel: minimumDurabilityLevel,
 			},
 			ConflictResolutionType: conflictResolutionType,
+			ReplicaIndexDisabled:   !replicaIndexes,
+			BucketType:             bucketType,
+			StorageBackend:         storageBackend,
```
Comment (Westwooo) [inline gateway/dataimpl/server_v1/bucketadminserver.go:207]: Why are these settings now immutable? Should they have been all along?

**Review by Westwooo — APPROVED** [review]
lgtm, just one question

File: gateway/dataimpl/server_v1/bucketadminserver.go (hunk 1 above)
Comment (chvck) [reply gateway/dataimpl/server_v1/bucketadminserver.go:207]: Yeah, the server doesn't allow changing these.

**Pull request merged**